zio-http/src/main/scala/zio/http/ZClient.scala

```diff
@@ -30,6 +30,7 @@ final case class ZClient[-Env, -In, +Err, +Out](
   url: URL,
   headers: Headers,
   sslConfig: Option[ClientSSLConfig],
+  proxy: Option[Proxy],
```

Comment:
You need to add a `def proxy` method that lets the user specify a proxy just like current exists with `def ssl`. You also need to add a test that uses this functionality.

Reply:
done